File: app/src/main/java/com/suzei/minote/ui/editor/EditorContract.java
package com.suzei.minote.ui.editor;

import com.suzei.minote.data.entity.Notes;
import com.suzei.minote.utils.ColorWheel;

public interface EditorContract {

    interface View {

        void setPresenter(Presenter presenter);

        void showNoteDetails(Notes note);

        void noteColor(int noteColor);

        void textColor(int textColor);

        void showNoteSave();

        void showColorWheel(String title, int initialColor, ColorWheel colorWheel);

        void showPasswordDialog();
    }

    interface Presenter {

        void start();

        void saveNote(String title,
                      String message,
                      String noteColor,
                      String textColor,
                      String password);

        void passwordDialog();

        void noteColorWheel(int initialColor);

        void textColorWheel(int initialColor);

    }

}

File: app/src/main/java/com/suzei/minote/ui/editor/EditorContract.kt
package com.suzei.minote.ui.editor

import com.suzei.minote.data.entity.Notes
import com.suzei.minote.utils.ColorWheel

interface EditorContract {

    interface View {

        fun setPresenter(presenter: Presenter)

        fun showNoteDetails(note: Notes)

        fun noteColor(noteColor: Int)

        fun textColor(textColor: Int)

        fun showNoteSave()

        fun showColorWheel(title: String, initialColor: Int, colorWheel: ColorWheel)

        fun showPasswordDialog()
    }

    interface Presenter {

        fun start()

        fun saveNote(title: String,
                     message: String,
                     noteColor: String,
                     textColor: String,
                     password: String?)

        fun passwordDialog()

        fun noteColorWheel(initialColor: Int)

        fun textColorWheel(initialColor: Int)

    }

}
